stdlib/public/runtime/DynamicCast.cpp

```diff
@@ -764,14 +770,18 @@ tryCastToAnyHashable(
   assert(cast<StructMetadata>(destType)->Description
          == &STRUCT_TYPE_DESCR_SYM(s11AnyHashable));
 
+  bool useCachedNSStringConformance = false;
   switch (srcType->getKind()) {
   case MetadataKind::ForeignClass: // CF -> String
   case MetadataKind::ObjCClassWrapper: { // Obj-C -> String
 #if SWIFT_OBJC_INTEROP
-    // TODO: Implement a fast path for NSString->AnyHashable casts.
-    // These are incredibly common because an NSDictionary with
-    // NSString keys is bridged by default to [AnyHashable:Any].
-    // Until this is implemented, fall through to the general case
+    auto cls = srcType;
+    do {
+      if (cls == getNSStringMetadata()) {
```

Comment:
```
 auto nsString = getNSStringMetadata();
 do {
    if (cls == nsString) {
        hashableConformance = getNSStringHashableConformance();
        break;
    }
    cls = _swift_class_getSuperclass(cls);
  } while (cls != nullptr);
  break;
```
This hoists the `getNSStringMetadata()` call out of the loop and does the hashable conformance fetch directly instead of setting a boolean to test later.